File: 백준/Silver/10866. 덱/덱.cc
#include<bits/stdc++.h>
using namespace std;
int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    deque<int> d;
    int n;
    cin >> n;
    while (n--) {
        string s;
        cin >> s;
        if (s == "push_front") {
            int k;
            cin >> k;
            d.push_front(k);
        }
        else if (s == "push_back") {
            int k;
            cin >> k;
            d.push_back(k);
        }
        else if (s == "pop_front") {
            if (d.empty()) cout << -1 << "\n";
            else {
                cout << d.front() << "\n";
                d.pop_front();
            }
        }
        else if (s == "pop_back") {
            if (d.empty()) cout << -1 << "\n";
            else {
                cout << d.back() << "\n";
                d.pop_back();
            }
        }
        else if (s == "size") {
            cout << d.size() << "\n";
        }
        else if (s == "empty") {
            cout << (int)d.empty() << "\n";
        }
        else if (s == "front") {
            if (d.empty()) cout << -1 << "\n";
            else cout << d.front() << "\n";
        }
        else if (s == "back") {
            if (d.empty()) cout << -1 << "\n";
            else cout << d.back() << "\n";
        }
    }
}

File: 백준/Silver/10866. 덱/덱.py
from collections import deque
import sys

input = sys.stdin.readline
n = int(input())
q = deque()
for i in range(n):
    a = list(input().split())
    if a[0]=='push_front':
        q.appendleft(int(a[1]))
    elif a[0]=='push_back':
        q.append(int(a[1]))
    elif a[0]=='pop_back':
        if not q:
            print(-1)
        else:
            print(q.pop())
    elif a[0]=='pop_front':
        if not q:
            print(-1)
        else:
            print(q.popleft())
    elif a[0]=='size':
        print(len(q))
    elif a[0]=='empty':
        if q:
            print(0)
        else:
            print(1)
    elif a[0]=='front':
        if not q:
            print(-1)
        else:
            print(q[0])
    elif a[0]=='back':
        if not q:
            print(-1)
        else:
            print(q[-1])